client/landing/gutenboarding/onboarding-block/edit.tsx

```diff
@@ -37,40 +44,81 @@ const OnboardingEdit: FunctionComponent< BlockEditProps< Attributes > > = () =>
 		window.scrollTo( 0, 0 );
 	}, [ pathname ] );
 
+	const canUseDesignSelection = useCallback( (): boolean => {
+		return ! ( ! siteVertical && ! siteTitle && ! wasVerticalSkipped );
+	}, [ siteTitle, siteVertical, wasVerticalSkipped ] );
+
+	const canUseStyleStep = useCallback( (): boolean => {
+		return !! selectedDesign && isEnabled( 'gutenboarding/style-preview' );
+	}, [ selectedDesign ] );
+
+	const canUseCreateSiteStep = useCallback( (): boolean => {
+		return isCreatingSite || isRedirecting;
+	}, [ isCreatingSite, isRedirecting ] );
+
+	const getLatestStepPath = (): string => {
+		if ( ! canUseDesignSelection() ) {
+			return makePath( Step.IntentGathering );
+		}
+		if ( ! canUseStyleStep ) {
+			return makePath( Step.DesignSelection );
+		}
+		return makePath( Step.IntentGathering );
+	};
+
+	useEffect( () => {
+		if (
+			! isCreatingSite &&
+			! newSite &&
+			currentUser &&
+			shouldTriggerCreate &&
+			canUseDesignSelection() &&
+			canUseStyleStep()
+		) {
+			createSite( currentUser.username, freeDomainSuggestion );
+		}
+	}, [
+		canUseDesignSelection,
+		canUseStyleStep,
+		createSite,
+		currentUser,
+		freeDomainSuggestion,
+		isCreatingSite,
+		newSite,
+		shouldTriggerCreate,
+	] );
+
 	return (
 		<div className="onboarding-block" data-vertical={ siteVertical?.label }>
 			{ isCreatingSite && (
-				<Redirect push={ replaceHistory ? undefined : true } to={ makePath( Step.CreateSite ) } />
+				<Redirect
+					push={ shouldTriggerCreate ? undefined : true }
+					to={ makePath( Step.CreateSite ) }
+				/>
 			) }
 			<Switch>
 				<Route exact path={ makePath( Step.IntentGathering ) }>
 					<AcquireIntent />
 				</Route>
 
 				<Route path={ makePath( Step.DesignSelection ) }>
-					{ ! siteVertical && ! siteTitle && ! wasVerticalSkipped ? (
-						<Redirect to={ makePath( Step.IntentGathering ) } />
-					) : (
+					{ canUseDesignSelection() ? (
 						<DesignSelector />
+					) : (
+						<Redirect to={ makePath( Step.IntentGathering ) } />
 					) }
 				</Route>
 
 				<Route path={ makePath( Step.Style ) }>
-					{
-						// Disable reason: Leave me alone, my nested ternaries are amazing ✨
```

Comment:
RIP amazing nested ternaries

Reply:
I was sad to see them go, I did love the comment!